Ginger/Ginger/UserControlsLib/ConsumerComboBox.xaml.cs

```diff
@@ -140,26 +144,38 @@ private static void OnSelectedConsumerChanged(DependencyObject d, DependencyProp
         }
 
 
-        public event PropertyChangedEventHandler PropertyChanged;
+        public event PropertyChangedEventHandler? PropertyChanged;
         public void OnPropertyChanged(string propertyName)
         {
-            PropertyChangedEventHandler handler = PropertyChanged;
+            PropertyChangedEventHandler? handler = PropertyChanged;
             if (handler != null)
             {
                 handler(this, new PropertyChangedEventArgs(propertyName));
             }
         }
 
-        private void ConsumerCheckBox_Click(object sender, RoutedEventArgs e)
+        private void ConsumerGrid_MouseUp(object sender, MouseButtonEventArgs e)
         {
-            CheckBox clickedBox = (CheckBox)sender;
+            Grid grid = (Grid)sender;
+            CheckBox checkBox = (CheckBox)grid.Children[0];
+            if (checkBox.IsChecked == null)
+                return;
 
-               
-                
-            
+            bool isChecked = (bool)checkBox.IsChecked;
+            checkBox.IsChecked = !isChecked;
+            if (isChecked)
+                checkBox.RaiseEvent(new RoutedEventArgs(CheckBox.UncheckedEvent));
+            else
+                checkBox.RaiseEvent(new RoutedEventArgs(CheckBox.CheckedEvent));
+
+            //prevent ComboBox dropdown from closing after clicking an item
+            e.Handled = true;
+        }
```

Comment:
Since we handle the click event for the entire ComboBox item (which is a Grid), it is possible that we only get a click on the Grid and not the child CheckBox. In order to handle that situation, we manually toggle the IsChecked property and after that to continue the remaining process as usual, we trigger the Checked/Unchecked event.